pennylane/devices/_qubit_device.py

```diff
@@ -758,7 +758,7 @@ def statistics(
             elif isinstance(m, (SampleMeasurement, StateMeasurement)):
                 result = self._measure(m, shot_range=shot_range, bin_size=bin_size)
 
-            elif m.return_type is not None:
+            elif m._shortname is not None:  # pylint: disable=protected-access
```

Comment:
```suggestion
            else:
```
Same as comment above. We shouldn't just silently ignore random objects.